src/confluent_kafka/schema_registry/avro.py

```diff
@@ -187,11 +187,20 @@ def __init__(self, schema_registry_client, schema_str,
         schema = _schema_loads(schema_str)
         schema_dict = loads(schema.schema_str)
         parsed_schema = parse_schema(schema_dict)
-        # The Avro spec states primitives have a name equal to their type
-        # i.e. {"type": "string"} has a name of string.
-        # This function does not comply.
-        # https://github.com/fastavro/fastavro/issues/415
-        schema_name = parsed_schema.get('name', schema_dict['type'])
+        
+        # if parsed_schema is a list, we have an Avro union and there
```

Comment:
If

Reply:
Put this comment within the if clause.

Reply:
👍 would be good, but not deal breaker for me for merging if it's the only holding things up.